lib/tlog/es_json_reader.c

```diff
@@ -364,6 +364,16 @@ tlog_es_json_reader_refill_array(struct tlog_es_json_reader *es_json_reader)
         goto cleanup;
     }
 
+    /* Allow kerberos (negotiate) authentication */
+    curl_easy_setopt(es_json_reader->curl, CURLOPT_HTTPAUTH, CURLAUTH_DIGEST | CURLAUTH_BASIC | CURLAUTH_GSSNEGOTIATE);
+
+    /* Read credentials from .netrc if present */
+    curl_easy_setopt(es_json_reader->curl, CURLOPT_NETRC, CURL_NETRC_OPTIONAL);
```

Comment:
We should check the return code of each curl library call
~~~
    rc = curl_easy_setopt(es_json_reader->curl, CURLOPT_WRITEFUNCTION,
                          tlog_es_json_reader_write_func);
    if (rc != CURLE_OK) {
        grc = TLOG_GRC_FROM(curl, rc);
        goto error;
    }
~~~